x-pack/elastic-agent/pkg/agent/transpiler/rules.go

```diff
@@ -225,7 +230,13 @@ type RemoveKeyRule struct {
 }
 
 // Apply applies remove key rule.
-func (r *RemoveKeyRule) Apply(_ AgentInfo, ast *AST) error {
+func (r *RemoveKeyRule) Apply(_ AgentInfo, ast *AST) (err error) {
+	defer func() {
+		if err != nil {
+			err = errors.New(err, "RemoveKeyRule")
```

Comment:
Pretty cryptic error message. It will make it easy for the software engineer to find where the error happened but will not be useful for an end user I think.